### PR TITLE
CORDA-3845: Update BC, log4j, slf4j (#6699)

### PR DESCRIPTION
**Pull request by rnicoll**
* CORDA-3845: Update BC to 1.64
* CORDA-3845: Upgraded log4j to 2.12.1
* We can remove the use of Manifests from the logging package so that when _it_ logs it doesn't error on the fact the stream was already closed by the default Java logger.
* Remove the logging package as a plugin
* latest BC version
* Remove old test
* Fix some rebased changes to log file handling
* Update slf4j too

Co-authored-by: Ryan Fowler <fowlerrr@users.noreply.github.com>
Co-authored-by: Adel El-Beik <adel.el-beik@r3.com>

### CODE REVIEW

**Review by fowlerrr — APPROVED** [review]
At least it works from here.  Have you done multiple builds?  Or was it failing regularly (and repeatedly) for 4.3?

**Comment by rnicoll** [comment]
Re-running to clear Sonatype status, and will let you know how that goes.

@mnesbit Can you confirm you're happy with this, please?

**Comment by rnicoll** [comment]
Oh and yes 4.3 was consistently failing every time with various errors I'm 90% certain were OOM underlying them somewhere.

File: constants.properties [hunk 1]
```diff
@@ -20,8 +20,8 @@ quasarClassifier=jdk8
 quasarVersion11=0.8.0_r3
 jdkClassifier11=jdk11
 proguardVersion=6.1.1
-bouncycastleVersion=1.61
-classgraphVersion=4.8.58
+bouncycastleVersion=1.66
+classgraphVersion=4.8.78
```
Comment (mnesbit) [inline constants.properties:24]: We are up to 4.8.90 on current branch, not sure what collateral that involves

Comment (adelel1) [reply constants.properties:24]: With 4.8.78 you wont be able to support '+' in CorDapp filenames, although this probably never worked in 58 either. If you go to 90 then you should change filterClasspathElements in JarScanningCordappLoader to...
        val scanResult = ClassGraph()
            .filterClasspathElementsByURL { elt -> elt == cordappJarPath.url }

i.e. filter now by URL.

**Comment by rnicoll** [comment]
My inclination is to use the patch as-written rather than trying to adapt other further fixes in, if that's okay with you @adelel1 ?

**Comment by adelel1** [comment]
> My inclination is to use the patch as-written rather than trying to adapt other further fixes in, if that's okay with you @adelel1 ?

Yes OK, makes sense.

**Pull request merged**